packages/@react-spectrum/dialog/docs/DialogTrigger.mdx

```diff
@@ -0,0 +1,464 @@
+
+```tsx example
+<DialogTrigger type="popover" placement="top" offset={-50}>
+  <ActionButton>Trigger</ActionButton>
+  <Dialog>
+    <Heading>The Heading</Heading>
+    <Divider />
+    <Content>
+      <Text>
+        This is a popover offset by an additional 50px above the trigger.
+      </Text>
+    </Content>
+  </Dialog>
+</DialogTrigger>
+
+<DialogTrigger type="popover" placement="top" crossOffset={100}>
+  <ActionButton>Trigger</ActionButton>
+  <Dialog>
+    <Heading>The Heading</Heading>
+    <Divider />
+    <Content>
+      <Text>
+        This is a popover offset by an additional 100px to the right of the trigger.
+      </Text>   
+    </Content>
+  </Dialog>
+</DialogTrigger>
+```
+
+### Flipping
+
+Applying `shouldFlip` to the DialogTrigger makes the Popover Dialog attempt to flip
+on its main axis in situations where the original placement would cause it to render out of view.
+
+```tsx example
+<DialogTrigger type="popover" placement="bottom" shouldFlip>
+  <ActionButton>Trigger</ActionButton>
+  <Dialog>
+    <Heading>The Heading</Heading>
+    <Divider />
+    <Content>
+      <Text>
+        This is a popover that will flip if it can't fully render below the button.
+      </Text>
+    </Content>
+  </Dialog>
+</DialogTrigger>
+```
+
+### Container Padding
+
+You can control the minimum padding required between the Popover Dialog and the 
+surrounding container via the `containerPadding` prop. This affects the positioning
+breakpoints that determine when the Dialog will attempt to flip.
+
+The example below will flip the Dialog from above the trigger button to below the trigger
+button if the Dialog cannot render fully while maintaining 50px of padding between itself and
+the top of the browser.
+```tsx example
+<DialogTrigger type="popover" shouldFlip position="top" containerPadding={50}>
+  <ActionButton>Trigger</ActionButton>
+  <Dialog>
+    <Heading>The Heading</Heading>
+    <Divider />
+    <Content>
+      <Text>
+        This is a popover.
+      </Text>
+    </Content>
+  </Dialog>
+</DialogTrigger>
+```
+
+## Events
+
+DialogTrigger accepts an `onOpenChange` handler which is triggered whenever the Dialog is opened or closed.
+
+The example below uses `onOpenChange` to update a separate `span` element with the current open state of the
+Dialog.
+
+```tsx example
+function Example() {
+  let [state, setState] = React.useState(false);
+
+  return (
+    <div>
+      <DialogTrigger type="popover" placement="top" onOpenChange={(isOpen) => setState(isOpen)}>
+        <ActionButton>Whispers</ActionButton>
+        <Dialog>
+          <Heading>Whispers and DMs</Heading>
+          <Divider />
+          <Content>
+            <Text>
+              You have 0 new messages.
+            </Text>
+          </Content>
+        </Dialog>
+      </DialogTrigger>
+      <span>Current open state: {state.toString()}</span>
```

Comment:
I borrowed this example for MenuTrigger and @MidnightCoder06 suggested I add space between the button and the span. I did `<span style={{'margin-left': '8px'}}>`